composer/core/state.py

```diff
@@ -295,11 +316,21 @@ def __init__(
             'scaler',
             'timestamp',
             'rank_zero_seed',
-            'current_metrics',
+            'train_metrics',
+            'eval_metrics',
             'run_name',
         ]
 
-        self.current_metrics: Dict[str, Dict[str, Any]] = {}
+        self.train_metrics: Dict[str, Metric] = {}
+        self.eval_metrics: Dict[str, Dict[str, Metric]] = {}
+
+    @property
+    def current_metrics(self):
+        warnings.warn(
+            DeprecationWarning(
+                'The ``current_metrics`` argument for a :class:`Trainer`. state is deprecated and will be removed in the future. Please use ``train_metrics`` and'
```

Comment:
DeprecationWarnings are not rendered with RST, so can just do:

The current_metrics property for Trainer.state is deprecated and will be removed in the future. Please use train_metrics or eval_metrics instead.